service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -180,7 +159,44 @@ public boolean updateWorkspace(UUID workspaceId, String name, String description
       params.addValue("description", description);
     }
 
-    return DbRetryUtils.retry(() -> updateWorkspaceHelper(workspaceId, params));
+    return updateWorkspaceHelper(workspaceId, params);
+  }
+
+  /**
+   * This is an open ended method for constructing the SQL update statement. To use it, build the
+   * parameter list making the param name equal to the column name you want to update. The method
+   * generates the column_name = :column_name list. It is an error if the params map is empty.
+   *
+   * @param workspaceId workspace identifier - not strictly necessarily, but an extra validation
+   * @param params sql parameters
+   */
+  private boolean updateWorkspaceHelper(UUID workspaceId, MapSqlParameterSource params) {
+    StringBuilder sb = new StringBuilder("UPDATE workspace SET ");
+
+    String[] parameterNames = params.getParameterNames();
+    if (parameterNames.length == 0) {
+      throw new MissingRequiredFieldException("Must specify some data to be updated.");
+    }
+    for (int i = 0; i < parameterNames.length; i++) {
```

Comment:
Bust a helper method for building the substring from the column list so at least this much won't need to be duplicated.